metrics/ci_integrations/test/source/jenkins/adapter/jenkins_source_client_adapter_test.dart

```diff
@@ -0,0 +1,597 @@
+
+        final result = adapter.fetchBuilds(jobName);
+
+        expect(result, completion(equals(expected)));
+      },
+    );
+
+    test(
+      ".fetchBuildsAfter() fetches builds which are not building",
+      () {
+        const build = BuildData(buildNumber: 1);
+        final jenkinsBuilds = [
+          createJenkinsBuild(buildNumber: 1),
+          createJenkinsBuild(buildNumber: 2),
+          createJenkinsBuild(buildNumber: 3, building: true),
+        ];
+        final expected = [createBuildData(buildNumber: 2)];
+
+        responses.addBuilds(jenkinsBuilds);
+
+        whenFetchBuilds().thenAnswer(responses.fetchBuilds);
+
+        final result = adapter.fetchBuildsAfter(jobName, build);
+
+        expect(result, completion(equals(expected)));
+      },
+    );
+
+    test(
+      ".fetchBuildsAfter() fetches a coverage for each build",
+      () async {
+        const build = BuildData(buildNumber: 1);
+        final jenkinsBuilds = [
+          createJenkinsBuild(buildNumber: 2),
+          createJenkinsBuild(buildNumber: 3, artifacts: [])
+        ];
+        const expected = [defaultCoverage, null];
+
+        responses.addBuilds(jenkinsBuilds);
+
+        whenFetchBuilds().thenAnswer(responses.fetchBuilds);
+
+        final list = await adapter.fetchBuildsAfter(jobName, build);
+        final coverages = list.map((buildData) => buildData.coverage).toList();
+
+        expect(coverages, equals(expected));
+      },
+    );
+
+    test(
+      ".fetchBuildsAfter() throws a StateError if fetching an artifact content fails for any of the given builds",
+      () {
+        const build = BuildData(buildNumber: 1);
+        const fileName = 'coverage-summary.json';
+        const relativePath = 'test/$fileName';
+        final jenkinsBuild = createJenkinsBuild(buildNumber: 2);
+        final nonExistingCoverageJenkinsBuild = createJenkinsBuild(
+          buildNumber: 3,
+          artifacts: const [
+            JenkinsBuildArtifact(
+              fileName: fileName,
+              relativePath: relativePath,
+            ),
+          ],
+        );
+        final jenkinsBuilds = [jenkinsBuild, nonExistingCoverageJenkinsBuild];
+
+        responses.addBuilds(jenkinsBuilds);
+
+        whenFetchArtifact(
+          buildUrlThat: equals(jenkinsBuild.url),
+          relativePathThat: anything,
+        ).thenAnswer(responses.artifact);
+
+        whenFetchArtifact(
+          buildUrlThat: equals(nonExistingCoverageJenkinsBuild.url),
+          relativePathThat: equals(relativePath),
+        ).thenAnswer((_) => responses.error<Map<String, dynamic>>());
+
+        when(jenkinsClientMock.fetchBuilds(
+          jobName,
+          limits: anyNamed('limits'),
+        )).thenAnswer(responses.fetchBuilds);
+
+        final result = adapter.fetchBuildsAfter(jobName, build);
+
+        expect(result, throwsStateError);
+      },
+    );
+
+    test(
+      ".fetchBuildsAfter() fetches new builds performed after the given build",
```

Comment:
It looks like we test mapping here, feels like this test could be simplified to highlight "after" component